### PR TITLE
Update project to look like a pub package

### PR DESCRIPTION
**Pull request by chinmaygarde**
* This version still requires placing the `.mojom.dart` generated file in the source. Working around this in an upcoming patch

### CODE REVIEW

File: lib/hello_world.mojom.dart [hunk 1]
```diff
@@ -0,0 +1,357 @@
+// Copyright 2014 The Chromium Authors. All rights reserved.
```
Comment (Hixie) [inline lib/hello_world.mojom.dart]: this "The Foo Authors" line should match the other file

File: lib/hello_world.mojom.dart [hunk 2]
```diff
@@ -0,0 +1,357 @@
...
+// found in the LICENSE file.
+
+library hello_world_mojom;
+
+import 'dart:async';
+
+import 'package:mojo/bindings.dart' as bindings;
+import 'package:mojo/core.dart' as core;
+
+
```
Comment (Hixie) [inline lib/hello_world.mojom.dart]: too many blank lines

**Comment by tvolkert** [comment]
lgtm

File: lib/hello_world.mojom.dart (hunk 1 above)
Comment (Hixie) [inline lib/hello_world.mojom.dart]: never mind the earlier comments here, this file is autogenerated

**Comment by tvolkert** [comment]
lgtm

File: tool/travis.sh [hunk 3]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+# Copyright (c) 2016, the Dart project authors.  Please see the AUTHORS file
```
Comment (Hixie) [inline tool/travis.sh]: the AUTHORS file you speak of is not the Dart project authors file...

File: lib/hello_world.mojom.dart (hunk 1 above)
Comment (chinmaygarde) [inline lib/hello_world.mojom.dart]: This file is generated by the tool. I couldnt get pub to package up files in directories outside lib. So as a temporary workaround, this is checked in.

File: tool/travis.sh [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+# for details. All rights reserved. Use of this source code is governed by a
+# BSD-style license that can be found in the LICENSE file.
+
+# Fast fail the script on failures.
+set -e
+
+# Verify that the libraries are error free.
```
Comment (Hixie) [inline tool/travis.sh]: not flutter analyze?

File: tool/travis.sh (hunk 3 above)
Comment (chinmaygarde) [inline tool/travis.sh]: Updating, was using the oss_project_template

File: test/all_test.dart [hunk 5]
```diff
@@ -0,0 +1,12 @@
+// Copyright (c) 2016, the Flutter project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:sample/sample.dart';
+import 'package:test/test.dart';
+
+main() {
+  group('A group of tests', () {
+    test('First Test', () {});
```
Comment (Hixie) [inline test/all_test.dart]: not much of a test :-P

File: src/hello_world_impl.cc [hunk 6]
```diff
@@ -7,6 +7,7 @@
 #include <sstream>
```
Comment (Hixie) [inline src/hello_world_impl.cc]: this file starts by mentioning Chromium authors...

**Comment by Hixie** [comment]
LGTM. Does the service actually print or echo hello world?

**Comment by chinmaygarde** [comment]
> Does the service actually print or echo hello world?

    HelloWorldProxy helloWorldProxy = new HelloWorldProxy.unbound();
    MojoShell.instance.connectToService(HelloWorld.serviceName, helloWorldProxy);
    print(await helloWorldProxy.ptr.sayHello("World"));

^ Will connect to the service and print the message from the native side

**Comment by chinmaygarde** [comment]
Authors updated. Still trying to figure out how to get the generated files in a pub package. I have just used the oss_project_template to start. About testing, it is not clear how testing would work without having to mock away all the native components.

**Pull request merged**